src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java

```diff
@@ -90,6 +114,28 @@ protected void finished(StepContext context) throws Exception {
 
 	}
 
+    private static final class ResourceVariableNameExpander extends EnvironmentExpander {
+        private static final long serialVersionUID = 1;
+        private final Map<String,String> overrides;
+
+        private ResourceVariableNameExpander(String resourceVariableName, List<String> resourceNames) {
+            this.overrides = new HashMap<>();
+            this.overrides.put(resourceVariableName, joinResourceNames(resourceNames));
+        }
+
+        private String joinResourceNames(List<String> resourceNames) {
+            StringBuilder resourceName = new StringBuilder();
+            for (String res : resourceNames) {
+                resourceName.append((resourceName.length() == 0) ? res : ", " + res);
```

Comment:
The delimiter should either be a comma or a space, not both. We should also make sure the delimiter cannot be used in the resource name.